plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -67,17 +69,37 @@ func NewSchedulerServer() *SchedulerServer {
 func (s *SchedulerServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.Port, "port", s.Port, "The port that the scheduler's http service runs on")
 	fs.Var(&s.Address, "address", "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
-	s.ClientConfig.QPS = 20.0
-	s.ClientConfig.Burst = 100
-	client.BindClientConfigFlags(fs, &s.ClientConfig)
 	fs.StringVar(&s.AlgorithmProvider, "algorithm_provider", s.AlgorithmProvider, "The scheduling algorithm provider to use")
 	fs.StringVar(&s.PolicyConfigFile, "policy_config_file", s.PolicyConfigFile, "File with scheduler policy configuration")
 	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
+	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
 }
 
 // Run runs the specified SchedulerServer.  This should never exit.
 func (s *SchedulerServer) Run(_ []string) error {
-	kubeClient, err := client.New(&s.ClientConfig)
+	var kubeconfig *client.Config = &client.Config{}
+	if s.Kubeconfig != "" {
```

Comment:
Do you actually get exactly the same behavior from this?

```
kubeconfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
    &clientcmd.ClientConfigLoadingRules{ExplicitPath: s.Kubeconfig}, 
    &clientcmd.ConfigOverrides{
        ClusterInfo: clientcmdapi.Cluster{Server: s.Master}
    }).ClientConfig()
if err != nil{
    return err
}
```

Not exactly an easy one-liner, but it does re-use all the existing logic.  See 
 1.  error on missing kubeconfig file: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/loader.go#L107
 1.  error on loading you're already catching.
 1.  overriding the host if a value is specified: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/client_config.go#L327